### PR TITLE
Issue-5886 New functions `gui.set_screen_position()` and `gui.screen_to_local()`

### LINKED ISSUE #5886: Functions for using screen positions in GUI component gui.set_screen_pos()
**Is your feature request related to a problem? Please describe (REQUIRED):**
For manipulations between game world and gui or between different gui components the engine should have API to convert cordinates using some universal cordinates system. 
At the moment we use screen coordinates as such universal coordinates we may use for conversion. 

**Describe the solution you'd like (REQUIRED):**
The engine has functions to get screen cordinates, but there are no functions in gui to use them.
I think the engine should have :
`gui.set_screen_pos()`
and
`local pos = gui.convert_screen_pos_to_node_pos(node, screen_pos)` (it's hard to find clear and short name here, pls suggest your variants).

**Describe alternatives you've considered (REQUIRED):**
I implemented these functions in the native extension, but I think it should be part of the gui.* API.
https://github.com/AGulev/set_screen_position

### PR DESCRIPTION
**Pull request by AGulev**
Two new function added to make it easier to operate nodes in GUI component:
- `gui.set_screen_position(node, screen_position)` - set screen position for the node;
- `local local_position = gui.screen_to_local(node, screen_position)` - convert screen position to a local position of the supplied node.

Also, it fixed a bug when the returned value of `gui.get_screen_position()` might be affected by pivot value.

Fixes https://github.com/defold/defold/issues/5886

### CODE REVIEW

File: engine/gui/src/gui.cpp [hunk 1]
```diff
@@ -1406,7 +1406,6 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
             c->m_SceneTraversalCache.m_Version = 0;
         }
 
-        Matrix4 node_transform;
```
Comment (AGulev) [inline engine/gui/src/gui.cpp:1409]: this variable is never used

File: engine/gui/src/gui.cpp [hunk 2]
```diff
@@ -3880,6 +3879,90 @@ Result DeleteDynamicTexture(HScene scene, const dmhash_t texture_hash)
...
+        InternalNode* n = GetNode(scene, node);
+        InternalNode* parent_node = 0x0;
+        if (n->m_ParentIndex != INVALID_INDEX)
+        {
+            parent_node = &scene->m_Nodes[n->m_ParentIndex];
+        }
+        SetScreenPosition(scene, n, parent_node, Vector3(screen_position));
+    }
+
     Result SetNodeParent(HScene scene, HNode node, HNode parent, bool keep_scene_transform)
```
Comment (AGulev) [inline engine/gui/src/gui.cpp:3966]: I got functionality from this function into two: `ScreenToLocalPosition` and `SetScreenPosition` (2 versions each, one external usage in `gui.h` and another one for internal usage)

Comment (JCash) [reply engine/gui/src/gui.cpp:3966]: Pls make the internal functions "static"

**Review by JCash — DISMISSED** [review]
I think it looks good.

It would feel a lot better if there were some tests added for these functions?

**Pull request merged**